src/sink/nats.rs

```diff
@@ -0,0 +1,298 @@
+            }))
+        } else {
+            Err("Nats offramp requires a configuration.".into())
+        }
+    }
+}
+
+#[async_trait::async_trait]
+impl Sink for Nats {
+    async fn on_event(
+        &mut self,
+        _input: &str,
+        codec: &mut dyn Codec,
+        _codec_map: &HashMap<String, Box<dyn Codec>>,
+        mut event: Event,
+    ) -> ResultVec {
+        let ingest_ns = event.ingest_ns;
+        let insight = event.insight_ack();
+        if let Some(connection) = &mut self.connection {
+            for (value, meta) in event.value_meta_iter() {
+                let encoded = codec.encode(value)?;
+                let processed =
+                    postprocess(self.postprocessors.as_mut_slice(), ingest_ns, encoded)?;
+                let meta_data = meta.get_object("nats");
+                let mut headers = None;
+                let mut reply = None;
+                if let Some(meta_data) = meta_data {
+                    headers = meta_data.get("headers");
+                    reply = meta_data.get("reply");
+                }
+                for payload in processed {
+                    // prepare message reply
+                    let mut message_reply = None;
+                    if let Some(reply) = reply {
+                        if let Some(reply_str) = reply.as_str() {
+                            message_reply = Some(reply_str);
+                        }
+                    } else if let Some(config_reply) = &self.config.reply {
+                        message_reply = Some(config_reply.as_str());
+                    }
+                    // prepare message headers
+                    let mut key_val: Vec<(&str, &str)> = Vec::new();
+                    if let Some(headers) = headers {
+                        if let Some(headers_obj) = headers.as_object() {
+                            for (key, val) in headers_obj.iter() {
+                                if let Some(array) = val.as_array() {
+                                    for ele in array.iter() {
+                                        if let Some(ele) = ele.as_str() {
+                                            key_val.push((key, ele));
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                    } else {
+                        for (key, val) in &self.config.headers {
+                            for ele in val.iter() {
+                                key_val.push((key.as_str(), ele.as_str()));
+                            }
+                        }
+                    }
+                    let key_val_len = key_val.len();
+                    let nats_headers = Headers::from_iter(key_val);
+                    let message_headers = if key_val_len > 0 {
+                        Some(&nats_headers)
+                    } else {
+                        None
+                    };
```

Comment:
You could possibly sijmplify this as:

```suggestion
                    let message_headers = if key_val.is_empty() {
                        None
                    } else {
                        Some(Headers::from_iter(key_val))
                    };
```

and then call `message_headers.as_ref()` in the call to publish